tests/fixtures.py

```diff
@@ -269,6 +282,130 @@ def inner(module_path, inventory_paths, basedir=None, tf_var_files=None,
   return inner
 
 
+def e2e_validator(module_path, extra_files, tf_var_files, basedir=None):
+  """Function running apply, plan and destroy to verify the case end to end
+
+  1. Tests whether apply does not return errors
+  2. Tests whether plan after apply is empty
+  3. Tests whether destroy does not return errors
+  """
+  module_path = _REPO_ROOT / module_path
+  with _prepare_root_module(module_path) as test_path:
+    binary = os.environ.get('TERRAFORM', 'terraform')
+    tf = tftest.TerraformTest(test_path, binary=binary)
+    extra_files = [(module_path / filename).resolve()
+                   for x in extra_files or []
+                   for filename in glob.glob(x, root_dir=module_path)]
+    tf.setup(extra_files=extra_files, upgrade=True)
+    tf_var_files = [(basedir / x).resolve() for x in tf_var_files or []]
+
+    try:
+      apply = tf.apply(tf_var_file=tf_var_files)
+      plan = tf.plan(output=True, tf_var_file=tf_var_files)
+      changes = {}
+      for resource_name, value in plan.resource_changes.items():
+        if value.get('change', {}).get('actions') != ['no-op']:
+          changes[resource_name] = value
+
+      # compare before with after to raise more meaningful failure to the user, i.e one
+      # that shows how resource will change
+      plan_before_state = dict((k, v['before']) for k, v in changes.items())
+      plan_after_state = dict((k, v['after']) for k, v in changes.items())
+
+      assert plan_before_state == plan_after_state, f'Plan not empty after apply for values'
+
+      plan_before_sensitive_state = dict(
+          (k, v['before_sensitive']) for k, v in changes.items())
+      plan_after_sensitive_state = dict(
+          (k, v['after_sensitive']) for k, v in changes.items())
+      assert plan_before_sensitive_state == plan_after_sensitive_state, f'Plan not empty after apply for sensitive values'
+
+      # If above did not fail, this should not either, but left as a safety check
+      assert changes == {}, f'Plan not empty for following resources: {", ".join(changes.keys())}'
+    finally:
+      destroy = tf.destroy(tf_var_file=tf_var_files)
+
+
+@pytest.fixture(name='e2e_validator')
+def e2e_validator_fixture(request):
+  """Return a function to run end-to-end test
+
+  In the returned function `basedir` becomes optional and it defaults
+  to the directory of the calling test
+
+  """
+
+  def inner(module_path: str, extra_files: list, tf_var_files: list,
+            basedir: os.PathLike = None):
+    if basedir is None:
+      basedir = Path(request.fspath).parent
+    return e2e_validator(module_path, extra_files, tf_var_files, basedir)
+
+  return inner
+
+
+@pytest.fixture(scope='session')
+def providers_tf():
```

Comment:
move this as a CLI argument to terraform

Reply:
I was confused with [`-backend-config`](https://developer.hashicorp.com/terraform/language/settings/backends/configuration#command-line-key-value-pairs)

Env var works too